File: day-01/java/Solution.java
import java.io.*;
import java.util.*;

public class Solution {
    private static ArrayList<Integer> totals = new ArrayList<>();

    public static void main(String[] args) {
        try {
            readInputFile();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

        System.out.println("Highest value: " + getHighestCals());
        System.out.println("Top 3: " + getTopThree());
    }

    private static void readInputFile() throws FileNotFoundException {
        Scanner in = new Scanner(new File("./day-01/input.txt"));

        int total = 0;
        while (in.hasNext()) {
            String line = in.nextLine();
            
            if (line.equals("")) {
                totals.add(total);
                total = 0;
            } else {
                total += Integer.parseInt(line);
            }
        }
    }

    private static int getHighestCals() {
        return Collections.max(totals);
    }

    private static int getTopThree() {
        Collections.sort(totals, Collections.reverseOrder());
        
        int topThree = 0;
        for (int i = 0; i < 3; i++) {
            topThree += totals.get(i);
        }
        return topThree;
    }
}

File: day-01/solution.py
def main():
    # Initialise a list to store each elfs total cals.
    totals = list()

    # Read the input file
    with open("./day-01/input.txt") as f:
        total = 0
        for row in f:
            # New line indiciators a new elf to total.
            if row == "\n":
                totals.append(total)
                total = 0
            else:
                # Increment total.
                total += int(row.rstrip("\n"))
        

    # Print the elf with the most cals.
    print(f"Max cals: {max(totals)}")

    # Get the sum of the top three elves cals
    print(f"Top 3 elves cals: {top_three_elfs(totals)}")


def top_three_elfs(totals: list) -> int:
    totals.sort(reverse = True)
    
    top3 = 0
    for i in range(3):
        top3 += totals[i]
    
    return top3


if __name__ == "__main__":
    main()
